src/sql/workbench/api/browser/mainThreadConnectionManagement.ts

```diff
@@ -146,8 +146,8 @@ export class MainThreadConnectionManagement extends Disposable implements MainTh
 			return undefined;
 		}
 
-		// Here we default to ConnectionType.editor which saves the connecton in the connection store by default
-		let connectionType = ConnectionType.editor;
+		// Here we default to ConnectionType.default which saves the connection in the connection store and server tree by default
+		let connectionType = ConnectionType.default;
```

Comment:
I think that's fine, since it's brining the information from connection completion options, whether or not to save this connection.. and setting `temporary` accordingly.

Reply:
What do you think setting it to temporary is doing? I just don't see it being needed with your change - we'll be respecting whatever values the user gave for whether the connection is saved or not.

Reply:
With 'temporary and 'editor' connection will NOT ve saved to server tree. So how I see it, if this logic determines it should make a connection 'temporary', that's a valid scenario, and should be respected as-is. Since with 'default' we're enabling saving to server tree view by default.

It's not applicable for my change, but there could be other paths that send 'connectionCompletionOptions' and set 'saveConnection' to false - it perhaps should set connection type as 'temporary' as per current behavior.

Reply:
Or do you mean because it's not considered here, it has not significance?
https://github.com/microsoft/azuredatastudio/blob/887053c60420d3d28a72d67fe88d856074f6f771/src/sql/workbench/services/connection/browser/connectionDialogService.ts#L429-L434

Reply:
Looking more into it, I see it being used to determine whether or not to show dashboard , that may still be valid.
https://github.com/microsoft/azuredatastudio/blob/887053c60420d3d28a72d67fe88d856074f6f771/src/sql/workbench/services/connection/browser/connectionDialogService.ts#L267-L268